src/generateTaskGraph.ts

```diff
@@ -0,0 +1,62 @@
+import { TopologicalGraph, Tasks, TaskId, PackageTaskDeps } from "./types";
+import { getTaskId, getPackageTaskFromId } from "./taskId";
+
+export function generateTaskGraph(
+  scope: string[],
+  targets: string[],
+  tasks: Tasks,
+  graph: TopologicalGraph
+): PackageTaskDeps {
+  const taskDeps: PackageTaskDeps = [];
+
+  const traversalQueue: TaskId[] = [];
+
+  for (const pkg of scope) {
+    for (const target of targets) {
+      traversalQueue.push(getTaskId(pkg, target));
+    }
+  }
+
+  const visited = new Set<TaskId>();
+
+  while (traversalQueue.length > 0) {
+    const taskId = traversalQueue.shift()!;
+    const [pkg, taskName] = getPackageTaskFromId(taskId);
+
+    if (!visited.has(taskId) && tasks.has(taskName)) {
+      visited.add(taskId);
+      const task = tasks.get(taskName)!;
+      const toTaskId = getTaskId(pkg, taskName);
+
+      if (task.topoDeps) {
```

Comment:
This API I'm not sure I like too much. In `lage`, my deps are specified like this: `{name:'build',deps:['^build','prepare']}`

Reply:
Though it's really not explicit enough for people to really understand what that is. So I am avoiding that in a library.

Reply:
Maybe {deps: ['prepare', parent('build')]}